api/hypershift/v1beta1/azure.go

```diff
@@ -433,8 +433,20 @@ type AzurePlatformSpec struct {
 	//
 	// +required
 	TenantID string `json:"tenantID"`
+
+	// Represents the encoding for the Azure Key Vault secret
+	// Plumbs the objectEncoding for secrets store provider - https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/master/website/content/en/getting-started/usage/_index.md
+	//
+	// +kubebuilder:validation:Enum:=utf-8;hex;base64
+	// +kubebuilder:default:=utf-8
+	// +kubebuilder:validation:Optional
+	ObjectEncoding ObjectEncodingFormat `json:"objectEncoding"`
```

Comment:
This is adding the attribute in `.platform.azure.objectEncoding`. If I understand this correctly, this means that conceptually this will apply to all keyvault secrets/certificates that are mounted using CSI driver, no matter where they come from.
Is this intentional? An alternative is in `.platform.azure.managedIdentities.controlPlane.managedIdentitiesKeyVault` ?

Take into account that other keyvault references are specified too in the spec (although I don't know if those end up being leveraged by CSI driver), like `.secretEncryption.kms.azure.activeKey`